test/scenarios/annotations/README.md

```diff
@@ -0,0 +1,77 @@
+# Annotated Requests
+
+## Purpose
+To create documentation for our interactions with pulp and candlepin across different workflows.  This allows the pulp and candlepin team to better understand how Katello is utilizing their respective service.
+
+## Components:
+
+1. Scenario test file: ./test/scenarios/scenario_test.rb 
+  * Currently houses all scenario tests.  Uses vcr to record reqeusts.
+2. Scenario VCR cassettes: ./test/fixtures/vcr_cassettes/scenarios/*.yml
+3. Scenario annotations: ./test/scenarios/annotations/*.yaml
+  * This provides options for matching a particular vcr request as well as a description of the request. 
+4. Scenario Rake tasks: ./lib/katello/tasks/annotate_scenarios.rake. Provides two rake tasks
+  * rake katello:find_unannotated_requests
+    * finds requests that are not annotated and suggests annotation entries
+  * rake katello:create_annotated_output
+    * creates the annotated request documentation
+    
+## Adding a new workflow
+
+1. Add a test in ./test/scenarios/scenario_test.rb 
+2. Delete the existing cassettes and re-record:
+```bash
+ cd ~/foreman/ #foreman git checkout
+ rm ../katello/test/fixtures/vcr_cassettes/scenarios/*.yml
+ rm ../katello/test/fixtures/vcr_cassettes/scenarios/support/*.yml
+ 
+ 
+ export mode=all
+ ktest ../katello/test/scenarios/scenario_test.rb
+ unset mode
+```
+3. Create a new scenario file: ./test/scenarios/annotations/my_workflow.yaml with contents:
+```yaml
+name: Performing Action XYZ
+cassette: cassette_name.yml
+annotations: []
+```
+Run the rake task to find unannotated requests:
+```
+rake katello:find_unannotated_requests
+```
+Example output
+```
+Performing Action XYZ
+Unmatched Annotations:
+Unmatched requests:
+---
+- method: post
+  path: "/some/path"
+  title: 
+  description: 
+```
+Simply copy and paste that into your annotation yaml file.  Edit, adding a description and title.  
```

Comment:
Are description and title required? If yes, can you add something that as part of checking for unannotated requests also "lints" the annotations to ensure these are filled out for each?